packages/mobile/src/identity/verification.ts

```diff
@@ -618,13 +623,55 @@ export function attestationCodeReceiver(
       return
     }
 
+    const allIssuers = attestations.map((a) => a.issuer)
+    let securityCode: string | null = null
+    let attestationCode
+
     try {
-      const code = extractAttestationCodeFromMessage(action.message)
-      if (!code) {
+      if (features.SHORT_VERIFICATION_CODES) {
+        securityCode = extractShortSecurityCodeMessage(action.message)
+        if (securityCode) {
+          const lookupHash = parseInt(securityCode[0], 10)
+          const lookupAttestations = attestations.filter(
+            (attestation) => hashAddressToSingleDigit(attestation.issuer) === lookupHash
+          )
+
+          for (const attestation of lookupAttestations) {
```

Comment:
Perhaps we can wrap the code creation and verification in a library/class. For protocols like this, it helps to not expose these details to the caller since a future code change can break it if the author doesn't understand how the complete picture.